src/plotrecipes.jl

```diff
@@ -1,60 +1,64 @@
 using DimensionalData: refdims_title
 
+const GeoDim = Union{GeoXDim,GeoYDim,GeoZDim}
+
 struct GeoPlot end
 
-# We only look at arrays with Lat/Lon here.
+# We only look at arrays with <:GeoXDim/<:GeoYDim here.
 # Otherwise they fall back to DimensionalData.jl recipes
 @recipe function f(A::AbstractGeoArray)
     A = GeoArray(A)
-    if all(hasdim(A, (Lat(), Lon())))
+    if all(hasdim(A, (GeoXDim, GeoYDim))) || all(hasdim(A, (GeoXDim, GeoZDim))) || all(hasdim(A, (GeoYDim, GeoZDim))) || hasdim(A, GeoZDim)
         # Heatmap or multiple heatmaps. Use GD recipes.
         GeoPlot(), prepare(A)
     else
-        # This is not a Lat/Lon heatmap. Fall back to DD recipes after reprojecting
+        # This is not a GeoXDim/GeoYDim heatmap. Fall back to DD recipes after reprojecting
         da = A |> GeoArray |> a -> DimArray(a; dims=_maybe_mapped(dims(a)))
         DimensionalData.DimensionalPlot(), da
     end
 end
 
 # Plot 3d arrays as multiple tiled plots
-@recipe function f(::GeoPlot, A::GeoArray{T,3,<:Tuple{<:Lat,<:Lon,D}}) where {T,D}
+@recipe function f(::GeoPlot, A::GeoArray{T,3,<:Tuple{<:GeoDim,<:GeoDim,D}}) where {T,D}
```

Comment:
This wont handle cases correctly where e.g the GeoXDim is the third dimension.

Reply:
Ah hmmm is this because we cannot assume the dimensions are in any order, e.g. X, Y, Z, T? How would you suggest I modify the function signature to handle all cases correctly?

Reply:
Yeah. Before I assumed Lat/Lon were the first two dims, but could be in either order. So the map always plots the right way.

Adding Z as well we probably can't assume anything now... so we can instead eagerly permute to the order that we want and only define one function signature with GeoPlot - the order will be fixed.